tests/compas/test_units_geometry.py

```diff
@@ -0,0 +1,314 @@
+        """Test Frame in unit-aware context."""
+        frame = Frame([1.0, 2.0, 3.0])
+        
+        # Frame can be part of unit-aware design data
+        design_data = {
+            'coordinate_frame': frame,
+            'scale': 1.0 * units.meter,
+            'tolerance': 0.01 * units.meter
+        }
+        
+        # Should serialize correctly
+        json_str = json.dumps(design_data, cls=DataEncoder)
+        reconstructed = json.loads(json_str, cls=DataDecoder)
+        
+        assert isinstance(reconstructed['coordinate_frame'], Frame)
+        assert hasattr(reconstructed['scale'], 'magnitude')
+        assert reconstructed['scale'].magnitude == 1.0
+
+
+@pytest.mark.skipif(not UNITS_AVAILABLE, reason="pint not available")
+class TestMeshWithUnitsWorkflow:
+    """Test Mesh in unit-aware workflows."""
+    
+    def test_mesh_with_unit_metadata(self):
+        """Test Mesh with unit metadata."""
+        # Create simple mesh
+        vertices = [[0.0, 0.0, 0.0], [1.0, 0.0, 0.0], [0.0, 1.0, 0.0]]
+        faces = [[0, 1, 2]]
+        mesh = Mesh.from_vertices_and_faces(vertices, faces)
+        
+        # Add unit-aware metadata
+        mesh_data = {
+            'mesh': mesh,
+            'units': 'meter',
+            'scale_factor': 1.0 * units.meter,
+            'material_thickness': 0.1 * units.meter,
+            'area': 0.5 * units.Quantity(1.0, 'meter^2')
+        }
+        
+        # Should serialize correctly
+        json_str = json.dumps(mesh_data, cls=DataEncoder)
+        reconstructed = json.loads(json_str, cls=DataDecoder)
+        
+        assert isinstance(reconstructed['mesh'], Mesh)
+        assert reconstructed['units'] == 'meter'
+        assert hasattr(reconstructed['scale_factor'], 'magnitude')
+        assert reconstructed['scale_factor'].magnitude == 1.0
+        assert hasattr(reconstructed['material_thickness'], 'magnitude')
+        assert reconstructed['material_thickness'].magnitude == 0.1
+    
+    def test_mesh_processing_workflow(self):
+        """Test mesh in processing workflow with units."""
+        # Create mesh with unit-aware attributes
+        mesh = Mesh()
+        
+        # Add vertices (plain coordinates)
+        v1 = mesh.add_vertex(x=0.0, y=0.0, z=0.0)
+        v2 = mesh.add_vertex(x=1.0, y=0.0, z=0.0)
+        v3 = mesh.add_vertex(x=0.0, y=1.0, z=0.0)
+        
+        # Add face
+        mesh.add_face([v1, v2, v3])
+        
+        # Add unit-aware attributes
+        mesh_analysis = {
+            'mesh': mesh,
+            'vertex_loads': [10.0 * units.Quantity(1.0, 'newton') for _ in mesh.vertices()],
+            'edge_lengths': [1.0 * units.meter for _ in mesh.edges()],
+            'face_areas': [0.5 * units.Quantity(1.0, 'meter^2') for _ in mesh.faces()]
+        }
+        
+        # Should serialize correctly
+        json_str = json.dumps(mesh_analysis, cls=DataEncoder)
+        reconstructed = json.loads(json_str, cls=DataDecoder)
```

Comment:
Same thing, add these attributes to the right location, the Mesh datastructure supports mesh, vertex, face and edge attributes